src/clarity-angular/datagrid/datagrid.html

```diff
@@ -36,7 +36,7 @@
 
     <!-- Custom placeholder overrides the default empty one -->
     <ng-content select="clr-dg-placeholder"></ng-content>
-    <clr-dg-placeholder *ngIf="!placeholder"></clr-dg-placeholder>
+    <clr-dg-placeholder *ngIf="!placeholder && items.displayed.length == 0"></clr-dg-placeholder>
```

Comment:
There is actually a bigger issue behind this, that I didn't point out explicitly: the purpose of this `*ngIf` is to always have a default placeholder when the user doesn't project one. So any sizing logic of the placeholder should never go here, but in the placeholder component itself. With this fix, the default placeholder and a custom one will behave completely different.

Now, regarding the "jump": one of the placeholder's purpose is to ensure the datagrid keeps the same height when moving to the last page. For instance, if you have a page size of 10 items, but the last page contains only 2, the placeholder will take the height of the missing 8 items to preserve the size of the datagrid.
If we remove this, the datagrid's height will change when moving to the last page, so whatever's under it will "jump" up. Then if you move back to the previous page, the content under it "jumps" back down.